src/test/java/uk/gov/hmcts/ccd/security/filters/ExceptionHandlingFilterTest.java

```diff
@@ -0,0 +1,59 @@
+package uk.gov.hmcts.ccd.security.filters;
+
+import org.apache.catalina.connector.ClientAbortException;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.junit.MockitoJUnitRunner;
+import org.springframework.mock.web.MockHttpServletRequest;
+import org.springframework.mock.web.MockHttpServletResponse;
+
+import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+
+@RunWith(MockitoJUnitRunner.class)
+public class ExceptionHandlingFilterTest {
+
+    private MockHttpServletRequest request;
+
+    private HttpServletResponse response;
+
+    @Mock
+    private FilterChain filterChain;
+
+    private ExceptionHandlingFilter filter;
+
+    @Before
+    public void setUp() {
+        filter = new ExceptionHandlingFilter();
+        request = new MockHttpServletRequest();
+        response = new MockHttpServletResponse();
+    }
+
+    @Test
+    public void shouldReturn502ResponseWhenClientAbortExceptionThrown() throws ServletException, IOException {
+        Mockito.doThrow(ClientAbortException.class)
+            .when(filterChain)
+            .doFilter(Mockito.eq(request), Mockito.eq(response));
+        filter.doFilterInternal(request, response, filterChain);
+
+        assertThat(response.getStatus()).isEqualTo(502);
+    }
+
+    @Test
+    public void shouldReturn500ResponseWhenAnyExceptionExceptClientAbortExceptionThrown()
+        throws ServletException, IOException {
+        Mockito.doThrow(NullPointerException.class)
+            .when(filterChain)
+            .doFilter(Mockito.eq(request), Mockito.eq(response));
+        filter.doFilterInternal(request, response, filterChain);
+
+        assertThat(response.getStatus()).isEqualTo(500);
+    }
```

Comment:
Is it worth checking that the expected error message is written to the log in these tests?  See RestExceptionHandlerTest class for examples of this being done.

Reply:
completed!

Reply:
completed!